### ISSUE TITLE
Routing does not match url-encoded path

### ISSUE THREAD

**Issue by PaulRivier** [scotty-web/scotty#262]
Hi there,

I can not find a way to match an URL encoded path such as `/page/P%C3%A2tes`
It seems to fail with both capture pattern, placeholders and regex.

Is this a bug ?

**Comment by chessai** [comment]
Can you post your code?

**Comment by PaulRivier** [comment]
Hello, thank you for your quick reply.

I messed my test cases before posting my original message, sorry about that. This is a more proper report, thank you for caring.

This works and matches /page/Pâtes
```
get "/page/:page" $ do
  p <- param "page"
  servePage p
```

But I need to match the rest of the url after page, not only first segment, so /page/recette/Pâtes would match with path=recette/Pâtes.  I could not find a clean way to get this behaviour, so I used the regex feature.

My regex is something like :

```
get (regex "^/page/(.+)")
  p <- param "1"
  servePage p
```

It does match correctly /page/Hello, but it is buggy on /page/Pâtes.  Basically, it will only match "P".

Interestingly, if regexp ends with "$" (like "^/page/(.+)$"), it does not match at all accentuated words.

Maybe the regex feature has some issues with multibytes chars. 

I would appreciate a lot to have a pattern to match the remaining of the path, like "/page/:~page" or so.

Thank you.

**Comment by ocramz** [comment]
@PaulRivier url-decoding parameters looks buggy indeed

**Comment by ocramz** [comment]
As pointed out by @jfraudeau this is indeed fixed by #302

**Comment by PaulRivier** [comment]
Thank you guys.
